### PR TITLE
Changes front end defaults

### LINKED ISSUE #4: Testing frameworks?
I decided I needed to write a JS library, and while doing so, I found myself investigating unit testing options for JS. I settled on `Jest` and it's great so far. But I wonder whether we wanted to specify our preferences for testing (e.g., pytest for python; Jest for JS; etc)

### PR DESCRIPTION
**Pull request by baccigalupi**
This also adds a breakout readme page for JS conventions.

### CODE REVIEW

File: JS.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+This is a conventions document for not getting into trouble in JS.
+
+In the future, we will have a little js pack repo for copying in a good set
+of standards into a new application.
+
+## React
+
+* Use `jsx` to build templates
+* Create templates that are as declarative as possible
```
Comment (jacobian) [inline JS.md:14]: This is probably a "Jacob is a React newbie" thing, but I don't really understand this. Is this something that someone more well-versed in React would understand? Or would it benefit from an example?

File: JS.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+* Put all state into the redux store
+* Even UI related state. Once this is a pattern it becomes easy
+* Don't use `jsx` in the connect files, instead create a new template
+* Extract logic for handlers into a handlers area
+* Extract logic reasoning about state into presenters
+
+## Testing
+
+* [jest](https://jestjs.io/) - for unit testing
+* [cypress](https://www.cypress.io/) - for acceptance testing
```
Comment (jacobian) [inline JS.md:29]: Should we be buying the paid Cypress thingie, or is the free version sufficient for our purposes?

**Review by jacobian — APPROVED** [review]
Looks good! I do have a few questions/comments (some inline, some below) but happy to merge with or without addressing 'em directly.

Extra questions:

- is Parcel working out OK or should we be switching to Webpack? (I hate Webpack, but also feel like Parcel might be not as flexible as we'd like)
- We should probably add a note to the backend part about being API-first and using DRF.
- Should we add a note about simpler options -- static sites, quick&dirty Django-only sites -- and when we might consider using them?

File: JS.md (hunk 1 above)
Comment (baccigalupi) [reply JS.md:14]: It seems like the rest of the repo is just a bullet list, so maybe I don't need to build out these conventions. It was hard to find a dividing line between describing in depth the kinds of things that go wrong with React and not enough information on the conventions.

I could definitely put in some examples too, but maybe this whole file could go or get slimmer.

Thoughts?

**Comment by baccigalupi** [comment]
Regarding Parcel and Webpack. I would rather stick with Parcel right now. There are a couple new systems coming out, and I would rather move forward than back. Webpack sucks so bad. Also, upgrading is not a blocker for right now. Sometimes there are weird caching things happening, but not as bad as I have seen with Rails, so 🤷 !

File: JS.md (hunk 1 above)
Comment (jacobian) [reply JS.md:14]: Yeah, dunno. Maybe just keep this as-is, but make it a topic for eng guild some day?

Comment (baccigalupi) [reply JS.md:14]: ✅

**Pull request merged**